amazon/ionbenchmark/benchmark_spec.py

```diff
@@ -176,12 +176,37 @@ def get_input_file_size(self):
 
     def get_data_object(self):
         """
-        Get the data object to be used for testing. Used for benchmarks that write data.
+        Get a list that holds all data objects to be used for testing. Used for benchmarks that write data.
         """
         if not self._data_object:
             loader = self.get_loader_dumper()
-            with open(self.get_input_file(), "rb") as fp:
-                self._data_object = loader.load(fp)
+            format_option = self.get_format()
```

Comment:
Why have all this branching here? we already have `loader`s for each type. Put the loader specific code there.

Reply:
Okay, I'll refactor them into a specific file there. I need to run the benchmark-cli again to ensure that the write/read process won't affect the performance (e.g., values generated by yield instead of returning an iterator directly).

Reply:
Refactor Json and Cbor, the performance are almost the same but the memory usage peak have a signaficant difference, I'm going to see why. 

Also since we dump the whole document for Ion to avoid unnecessary symbol table write, the Ion_load_dump needs to be changed too.

Reply:
I wonder if it is due to building the list?

Reply:
I'm also not sure why IonLoadDump needs to change. My understanding is that we're not flushing the writer so it will just buffer until it closes, then flush the symbol table and values. Right?

Reply:
Re:
> I'm also not sure why IonLoadDump needs to change. My understanding is that we're not flushing the writer so it will just buffer until it closes, then flush the symbol table and values. Right?

I saw a 20% difference between dumps each top-level object and dump the whole document. I'll try to see why it takes shorter when repeatedly calling dump